src/main/java/com/google/devtools/build/lib/buildeventservice/BuildEventServiceUploader.java

```diff
@@ -449,7 +449,16 @@ private void publishBuildEvents()
                       buildEvent.getSequenceNumber(),
                       buildEvent.getCreationTime(),
                       Any.pack(serializedRegularBuildEvent));
+              ackQueue.addLast(new SendRequestBuildEventCommand(request));
+              streamContext.sendOverStream(request);
+            }
+            break;
 
+          case SEND_REQUEST_BUILD_EVENT:
```

Comment:
It turns out there was a subtle bug here. In this case, when we are re-sending the last event, we need to treat this as if we are in `case SEND_LAST_BUILD_EVENT`, by closing the stream:

```java
              streamContext.sendOverStream(request);
              streamContext.halfCloseStream();
              halfCloseFuture.set(null);
```